### ISSUE TITLE
Reactivity is missing when createContextStore is used

### ISSUE THREAD

**Issue by iongion** [ctrlplusb/easy-peasy#907]
I made a demo here

<https://codesandbox.io/p/devbox/easy-peasy-v6-devbox-55sgc7>

Inside `./src/components/GraphicsEditor/Model.tsx` there are two ways to define the model store, either using

```typescript
export const EditorStore = createContextStore(EditorModel);
```

or with an equivalent using the typed hooks

```typescript
export const EditorModelStore = createStore<IEditorStore>(EditorModel);
const typedHooks = createTypedHooks<IEditorStore>();
export const useStoreActions = typedHooks.useStoreActions;
export const useStoreDispatch = typedHooks.useStoreDispatch;
export const useStoreState = typedHooks.useStoreState;
export const EditorStore = {
  Provider: ({ children }: any) => {
    return <StoreProvider store={EditorModelStore}>{children}</StoreProvider>;
  },
  useStoreActions,
  useStoreDispatch,
  useStoreState,
  useStore: () => EditorModelStore,
};
```

If one clicks on 
![image](https://github.com/ctrlplusb/easy-peasy/assets/867609/3b895e17-e2ec-4b0f-932f-d83d82dc3019)

1. when using the `createContextStore` it should append items in the array - it does so, but the UI does not react to it.
2. if the alternative approach is used, then it will append items and the UI will update properly

**Comment by jmyrland** [comment]
Hi @iongion !

Thanks for the detailed sandbox! I'll take a quick look at this and come back to you 👍

**Comment by jmyrland** [comment]
I forked your sandbox, and it seems that you have used multiple `EditorStore.Provider`s. This does not work.

You have use a single instance of this. So, to make your example work, I moved `EditorStore.Provider` into the `App`, wrapping both the `EditorToolbar` and `Editor`. 

```jsx
      <EditorStore.Provider>
        <EditorToolbar />
        <Editor />
      </EditorStore.Provider>
```

Is this the expected behaviour?
![chrome_mBAEXdLpHt](https://github.com/ctrlplusb/easy-peasy/assets/837651/414733da-f7a4-4902-9be6-37b9af5a36a5)
